recipes-extended/xen/files/libxl-RFC-fixes.patch

```diff
@@ -0,0 +1,76 @@
+CHANGELOG
+################################################################################
+Authors:
+Martin Osterloh <osterlohm@ainfosec.com>
+
+################################################################################
+REMOVAL
+################################################################################
+If the original RFC ever gets merged upstream, these issues will probably be
+fixed.
+
+################################################################################
+UPSTREAM PLAN
+################################################################################
+If we decide to initiate a third Linux stubdomain, this patch should be included
+
+################################################################################
+INTERNAL DEPENDENCIES
+################################################################################
+libxl-RFC-*.patch
+
+################################################################################
+PATCHES
+################################################################################
+Index: xen-4.6.1/tools/libxl/libxl_dm.c
+===================================================================
+--- xen-4.6.1.orig/tools/libxl/libxl_dm.c
++++ xen-4.6.1/tools/libxl/libxl_dm.c
+@@ -1243,12 +1243,14 @@ static void libxl__dm_vifs_from_hvm_gues
+                                     libxl_domain_config * const guest_config,
+                                     libxl_domain_config *dm_config)
+ {
++    libxl_ctx *ctx = libxl__gc_owner(gc);
+     int i, nr = guest_config->num_nics;
+ 
+     GCNEW_ARRAY(dm_config->nics, nr);
+ 
+     for (i=0; i<nr; i++) {
+-        dm_config->nics[i] = guest_config->nics[i];
++        libxl_device_nic_init(&dm_config->nics[i]);
++        libxl_device_nic_copy(ctx, &dm_config->nics[i], &guest_config->nics[i]);
+         dm_config->nics[i].nictype = LIBXL_NIC_TYPE_VIF;
+         if (dm_config->nics[i].ifname)
+             dm_config->nics[i].ifname = GCSPRINTF("%s" TAP_DEVICE_SUFFIX,
+@@ -1514,7 +1516,7 @@ void libxl__spawn_stub_dm(libxl__egc *eg
+                    "%d", guest_domid);
+     if (guest_config->b_info.stubdomain_version == LIBXL_STUBDOMAIN_VERSION_LINUX) {
+         /* qemu-xen is used as a dm in the stubdomain, so we set the bios
+-         * accroding to this */
++         * according to this */
+         libxl__xs_write(gc, XBT_NULL,
+                         libxl__sprintf(gc, "%s/hvmloader/bios",
+                                        libxl__xs_get_dompath(gc, guest_domid)),
+@@ -1711,7 +1713,10 @@ static void stub_dm_watch_event(libxl__e
+         if (dm_domid) {
+             sdss->dis.ao = sdss->dm.spawn.ao;
+             sdss->dis.domid = dm_domid;
+-            sdss->dis.callback = spaw_stubdom_pvqemu_destroy_cb;
++            /* FIXME: the following line references a non-existent function,
++             *   even without the typo. Commenting out for now.
++             */
++            /* sdss->dis.callback = spaw_stubdom_pvqemu_destroy_cb; */
```

Comment:
The whole block actually gets removed later (https://github.com/OpenXT/xenclient-oe/pull/473/files#diff-2755bad710d9aa9a2a9ff34e6783cfdcR134).
This patch is strictly to fix typos/incoherence found in the RFC code.
This would only need to be figured out if we were using a qdisk backend for the stubdom.